Mage.Sets/src/mage/cards/c/CarmenCruelSkymarcher.java

```diff
@@ -0,0 +1,108 @@
+package mage.cards.c;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.AttacksTriggeredAbility;
+import mage.abilities.effects.common.GainLifeEffect;
+import mage.abilities.effects.common.ReturnFromGraveyardToBattlefieldTargetEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.cards.Card;
+import mage.constants.*;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.counters.CounterType;
+import mage.filter.FilterCard;
+import mage.filter.common.FilterPermanentCard;
+import mage.filter.predicate.ObjectSourcePlayer;
+import mage.filter.predicate.ObjectSourcePlayerPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCardInYourGraveyard;
+
+/**
+ *
+ * @author skaspels
+ */
+public final class CarmenCruelSkymarcher extends CardImpl {
+
+    private static final FilterCard filter = new FilterPermanentCard(
+            " a permanent from your graveyard with mana value less than or equal to {this}'s power");
+    static {
+        filter.add(CarmenPredicate.instance);
+    }
+    public CarmenCruelSkymarcher(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{W}{B}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.VAMPIRE);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Whenever a player sacrifices a permanent, put a +1/+1 counter on Carmen, Cruel Skymarcher and you gain 1 life.
+        this.addAbility(new CarmenTriggeredAbility());
+
+        // Whenever Carmen attacks, return up to one target permanent card with mana value less than or equal to Carmen's power from your graveyard to the battlefield.
+        Ability ability = new AttacksTriggeredAbility(new ReturnFromGraveyardToBattlefieldTargetEffect());
+        ability.addTarget(new TargetCardInYourGraveyard(0,1,filter));
+        this.addAbility(ability);
+    }
+
+    private CarmenCruelSkymarcher(final CarmenCruelSkymarcher card) {
+        super(card);
+    }
+
+    @Override
+    public CarmenCruelSkymarcher copy() {
+        return new CarmenCruelSkymarcher(this);
+    }
+}
+
+class CarmenTriggeredAbility extends TriggeredAbilityImpl {
```

Comment:
this is okay as is for now but I'm making a note to clean up this and other cards to common classes (we have SacrificeAllTriggeredAbility and SacrificePermanentTriggeredAbility)